src/Http/Http.Extensions/src/HttpMetricsEndpointConventionBuilderExtensions.cs

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Builder;
+
+/// <summary>
+/// HTTP metrics extension methods for <see cref="IEndpointConventionBuilder"/>.
+/// </summary>
+public static class HttpMetricsEndpointConventionBuilderExtensions
+{
+    public static IEndpointConventionBuilder DisableHttpMetrics(this IEndpointConventionBuilder builder)
+    {
+        builder.Add(b => b.Metadata.Add(new DisableHttpMetricsAttribute()));
```

Comment:
Can this metadata be stored and be reused?

Reply:
We don't do that for other metadata. I'll leave it as is, and reusing/caching metadata can be considered more holistically in another issue if desired.